### PR TITLE
LibWebView+LibGUI+WindowServer: Implement drag-and-drop for OutOfProcessWebView 

### LINKED ISSUE #25417: Browser: Implement drag-n-drop
LibWeb gained drag-n-drop support in #25243. It's not hooked up in Browser yet. Need to do something similar to #25401 in Userland/Applications/Browser. (...assuming LibGUI has drag-n-drop support already.)

### PR DESCRIPTION
**Pull request by trflynn89**
Fixes #25417

[Screencast_20241202_090929.webm](https://github.com/user-attachments/assets/e3154129-f82b-45b5-ba5d-1bdd26e0072a)

### CODE REVIEW

File: Userland/Libraries/LibWebView/OutOfProcessWebView.cpp [hunk 1]
```diff
@@ -295,6 +328,54 @@ void OutOfProcessWebView::enqueue_native_event(Web::MouseEvent::Type type, GUI::
...
+{
+    auto position = to_content_position(event.position()).to_type<Web::DevicePixels>();
+    auto screen_position = (event.position() + (window()->position() + relative_position())).to_type<Web::DevicePixels>();
+
+    Vector<Web::HTML::SelectedFile> files;
+    OwnPtr<DragData> chrome_data;
+
+    auto button = static_cast<Web::UIEvents::MouseButton>(to_underlying(event.button()));
+    auto buttons = static_cast<Web::UIEvents::MouseButton>(event.buttons());
+    auto modifiers = static_cast<Web::UIEvents::KeyModifier>(event.modifiers());
```
Comment (nico) [inline Userland/Libraries/LibWebView/OutOfProcessWebView.cpp:350]: `Web::UIEvents::KeyModifier` is the same type at the moment, abut `Web::UIEvents::MouseButton` and `GUI::MouseButton` are distinct and just happen to line up. I suppose that's alright, but it's a bit wobbly :)

File: Userland/Libraries/LibWebView/OutOfProcessWebView.cpp [hunk 2]
```diff
@@ -295,6 +328,54 @@ void OutOfProcessWebView::enqueue_native_event(Web::MouseEvent::Type type, GUI::
...
+
+    if (type == Web::DragEvent::Type::DragStart) {
+        VERIFY(event.mime_data().has_urls());
+
+        for (auto const& url : event.mime_data().urls()) {
+            auto file_path = URL::percent_decode(url.serialize_path());
+
+            if (auto file = Web::HTML::SelectedFile::from_file_path(file_path); file.is_error())
+                warnln("Unable to open file {}: {}", file_path, file.error());
```
Comment (nico) [inline Userland/Libraries/LibWebView/OutOfProcessWebView.cpp:359]: should maybe say "web drag" in the text somewhere

File: Userland/Libraries/LibWebView/OutOfProcessWebView.cpp [hunk 3]
```diff
@@ -295,6 +328,54 @@ void OutOfProcessWebView::enqueue_native_event(Web::MouseEvent::Type type, GUI::
...
+            else
+                files.append(file.release_value());
+        }
+    } else if (type == Web::DragEvent::Type::Drop) {
+        chrome_data = make<DragData>(event);
+    }
```
Comment (nico) [inline Userland/Libraries/LibWebView/OutOfProcessWebView.cpp:365]: maybe `else VERIFY(type == DragMove)`

**Review by nico — APPROVED** [review]
Super cool!

Fine as-is, do with the below suggestions what you want :)

File: Userland/Libraries/LibWebView/OutOfProcessWebView.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibWebView/OutOfProcessWebView.cpp]: Yeah, I had just copied these from the mouse event code above, but I agree, I'd rather these be explicit. I added a commit to add conversions & use them in mouse events.

**Pull request merged**